### ISSUE TITLE
Array.filter is not a function

### ISSUE THREAD

**Issue by skddc** [contentful-labs/contentful-space-sync#13]
```
> Getting content from source space 
ERR! Failed with
 TypeError: Array.filter is not a function
ERR! Failed with
     at filterType (/home/basti/.node_modules/lib/node_modules/contentful-space-sync/dist/get/get-source-space.js:69:16)
```

This happens both with the installed binary from npm as well as when running `node dist/space-sync.js` directly.

**Comment by skddc** [comment]
Uh, when I use `babel-node` to run `dist/space-sync.js` it fails with a different error, which is rather weird:

```
> Getting content from source space 
ERR! Failed with
 DefaultLocaleNotDeletable: DELETE https://api.contentful.com:443/spaces/nwszylcdhgm6/locales/1TKFHYy0NBZKqL0PKj3sP4?access_token=0db1c6b9cfc43fcfd35a3404f050fa2a00dccaa4e6a
11c8414da7386010e1e4d Cannot delete a default locale. {"sys":{"type":"Error","id":"DefaultLocaleNotDeletable"},"requestId":"9e1-607131206","message":"Cannot delete a default
 locale."}
```

It tries to *delete* the default locale when *getting* the source space?!

**Comment by trodrigues** [comment]
Hi there,

regarding the first issue, what version of node are you running this on?

As for the second, api.contentful.com is used only to send data to the destination space. Is nwszylcdhgm6 the ID of the space you are using as the source space?

Also, I see you've pasted the whole URL including the access_token, which if this is the management token is probably not a good idea to have it out in the open. If you'd like to show me your configuration and more details that can be sensitive feel free to contact me directly by email at tiago@contentful.com

**Comment by skddc** [comment]
> regarding the first issue, what version of node are you running this on?

v5.0.0

> As for the second, api.contentful.com is used only to send data to the destination space. Is nwszylcdhgm6 the ID of the space you are using as the source space?

Ah, you're right. That was destination space.

> Also, I see you've pasted the whole URL including the access_token, which if this is the management token is probably not a good idea to have it out in the open.

Ugh, I will probably never get used to the fact that you use access tokens as URL params, so they're visible everywhere you can see requests. Thanks for the hint, already deleted.

**Comment by trodrigues** [comment]
> v5.0.0

Right! That's out! I think I haven't tested it yet on that version, so I'll give it a try. There might be some not obvious weirdness with ES6 and the compiled babel code.

> Ah, you're right. That was destination space.

So what happens essentially is, as we don't get the locales and content types via the sync endpoint, we don't know exactly which ones were deleted and/or updated. So what the script does is, it gets all the CTs/locales on the source and destination space, and in the destination space deletes the ones that stopped existing in the source space.

I've written this in the readme but let me know if that wasn't entirely obvious and I'll try to improve it.

Regarding as to why this was trying to delete the default locale, and with the explanation above: did you make changes to the locales in the destination space? Or did you change which locale is the default in the source space?

> Ugh, I will probably never get used to the fact that Contentful uses access tokens as URL params, so they're visible everywhere you can see requests. Thanks for the hint, already deleted.

Not always, we can also send them as headers. In fact, that's a good idea for the upcoming rewrite of the contentful-management.js library, so I'll write it down.

**Comment by skddc** [comment]
> Regarding as to why this was trying to delete the default locale, and with the explanation above: did you make changes to the locales in the destination space? Or did you change which locale is the default in the source space?

I'm not sure anymore. I did fix a locale issue afterwards where en-US was set as default for a space that was supposed to use sth else, so that should be the explanation then.

> Not always, we can also send them as headers. In fact, that's a good idea for the upcoming rewrite of the contentful-management.js library, so I'll write it down.

The delivery JS client does this as well, and it'd be great to change it there, too. ;)

**Comment by SMillerDev** [comment]
I have the same issue so I thought I'd post some more log that I get to hopefully make debug eassier.

   > Contentful Space Sync:
Let's sync some content across spaces!
 
> No previous sync token found or fresh sync requested.
  Synchronizing fresh content from bu3jhz3mxna4 to 9ekxodx7z4ou
   
> Getting content from source space 
ERR! Failed with
 TypeError: Array.filter is not a function
ERR! Failed with
     at filterType (/usr/local/lib/node_modules/contentful-space-sync/dist/get/get-source-space.js:69:16)
ERR! Failed with
     at /usr/local/lib/node_modules/contentful-space-sync/dist/get/get-source-space.js:33:48
ERR! Failed with
     at lib$es6$promise$$internal$$tryCatch (/usr/local/lib/node_modules/contentful-space-sync/node_modules/contentful-management/node_modules/es6-promise/dist/es6-promise.js:331:16)
ERR! Failed with
     at lib$es6$promise$$internal$$invokeCallback (/usr/local/lib/node_modules/contentful-space-sync/node_modules/contentful-management/node_modules/es6-promise/dist/es6-promise.js:343:17)
ERR! Failed with
     at lib$es6$promise$$internal$$publish (/usr/local/lib/node_modules/contentful-space-sync/node_modules/contentful-management/node_modules/es6-promise/dist/es6-promise.js:314:11)
ERR! Failed with
     at lib$es6$promise$asap$$flush (/usr/local/lib/node_modules/contentful-space-sync/node_modules/contentful-management/node_modules/es6-promise/dist/es6-promise.js:125:9)
ERR! Failed with
     at doNTCallback0 (node.js:428:9)
ERR! Failed with
     at process._tickCallback (node.js:357:13)
ERR! Failed with
  [TypeError: Array.filter is not a function]

**Comment by trodrigues** [comment]
Essentially this was due to babel-polyfill not being included, and I wrongly assumed the babel compiler would take care of that for me.

Issue is fixed in 1.2.1, just released.

**Comment by skddc** [comment]
Thanks!
